### ISSUE TITLE
Dynamic property of NestedAttribute not supported in NEST 5.5.0?

### ISSUE THREAD

**Issue by jo-rah** [elastic/elasticsearch-net#2919]
We want to upgrade NEST from 2.3.1 to 5.5.0, but have encountered a problem with having strict dynamic mapping of nested objects.

This is allowed with NEST 2.3.1:
```
        // ChildObject
        [Nested(Name = "ChildObject", Dynamic = DynamicMapping.Strict)]
        public ElasticsearchDynamicChildObjectBase<T> ChildObject { get; set; }
```

But with 5.5.0, the Dynamic property is not a public implementation of the NestedAttribute class and the compiler gives
`Error CS0246 The type or namespace name 'Dynamic' could not be found (are you missing a using directive or an assembly reference?)
`
The declaration of Nest.NestedAttribute is
`public class NestedAttribute : ObjectAttribute, INestedProperty, IObjectProperty, ICoreProperty, IProperty, IFieldMapping`

But the implementation of the Dynamic property has changed to an explicit interface implementation of IObjectProperty:
`Union<bool, DynamicMapping> IObjectProperty.Dynamic { get; set; }`

See https://github.com/elastic/elasticsearch-net/pull/2337/files#diff-7df79bafe8d65cbd85e726897b080c2e. Could row #15 in ObjectAttribute.cs have been removed by mistake?
 
This consequence of this is that it's no longer possible to set this property in the Attribute. I have searched, but not been able to figure out to set this in NEST 5.5.0. Maybe the strict mapping now is inherited from the index where we use code like this?
 ```
       var response = ec.CreateIndex(indexName, index => index.Mappings(descriptor => descriptor
						 .Map<ElasticsearchObjectBase>(mapping => mapping
						 .Dynamic(DynamicMapping.Strict)

```
In short, how do I get strict mapping of nested objects using NEST 5.5.0?

**Comment by Mpdreamz** [comment]
It's no longer possible to map this from the attributes. Because attributes are limited in what they can express we locked down what you can do with them to the bare minimum on purpose. 

You will have to map it explicitly using 

``csharp
.Nested<Developer>(n => n
	.Name(p => p.LeadDeveloper)
	.Dynamic(DynamicMapping.Strict)
	.Properties(pps => pps
		.Ip(i => i
			.Name(p => p.IpAddress)
		)
	)
);
```

Or you can use a visitor that sets it for `Nested`.

https://www.elastic.co/guide/en/elasticsearch/client/net-api/current/visitor-pattern-mapping.html

Hope this helps.
